src/mod_tools.cpp

```diff
@@ -174,15 +174,17 @@ void GameModeManager_t::Tutorial_t::readFromFile()
 		std::string inputPath = PHYSFS_getRealDir("/data/tutorial_strings.json");
 		inputPath.append("/data/tutorial_strings.json");
 
-		FILE* fp = fopen(inputPath.c_str(), "rb");
+		File* fp = FileIO::open(inputPath.c_str(), "rb");
 		if ( !fp )
 		{
 			printlog("[JSON]: Error: Could not locate json file %s", inputPath.c_str());
 			return;
 		}
 		char buf[65536];
-		rapidjson::FileReadStream is(fp, buf, sizeof(buf));
-		fclose(fp);
+		int count = fp->read(buf, sizeof(buf[0]), sizeof(buf));
+		buf[count] = '\0';
```

Comment:
No check that count < buffer size?

Reply:
Yeah I suppose that could be added, right now there's just an implicit expectation that the file will be <64kb. given they are just a few KBs right now, such an expectation seems reasonable...

Reply:
Ya...one big thing I can think of is a malicious attack, but that isn't the biggest worry for this sort of program.

The other one is if we give jsons to modders to modify, we should have an explicit rejection of files they are making too big, rather than segfaulting (or worse: silently accepting the bad juju)

Reply:
hold on, I went in to fix this and realized that fp->read() cannot overfill the buffer since it takes the size as the last parameter.

so the worst case scenario is that the json will be malformed, but that won't cause any serious error.

Reply:
Ah, that's good. A malformed JSON is much better than a hard crash :)

Reply:
Ideally we'd still hit the user with a proper error so they know why their JSON is "malformed"...